### PR TITLE
Feature/update readme changelog

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,22 @@
...
+* Refactor secret model.
+* Upgrade to Gradle 7.6.
+* Update web3 secret challenge.
+* Remove TLS context on server.
+* Remove `GET /secrets` endpoints.
+* Support Gramine framework for TEE tasks.
+* Add `GET /up` client method in iexec-sms-library.
+* Remove non-TEE workflow.
+* Return a same `SmsClient` from the `SmsClientProvider` of iexec-sms-library when calling a same SMS URL.
+* Improve code quality.
```
Comment (mcornaton) [inline CHANGELOG.md:19]: Could we define an order for changelog's lines? We can have something like:
- Dependencies updates
- New features
- Bug fixes
- Code quality (= refactor, documentation, ...)

File: README.md [hunk 2]
```diff
@@ -16,58 +21,72 @@ The iExec Secret Management Service (SMS) stores user secrets and provisions the
 
 ## Configuration
 
-The iExec Secret Management Service is available as an OCI image on [Docker Hub](https://hub.docker.com/r/iexechub/iexec-sms/tags).
+The _iExec Secret Management Service_ is available as an OCI image on [Docker Hub](https://hub.docker.com/r/iexechub/iexec-sms/tags).
+
+One _iExec Secret Management Service_ instance supports only one TEE framework kind of iExec TEE tasks.
```
Comment (mcornaton) [inline README.md:26]:
```suggestion
A single _iExec Secret Management Service_ instance supports a single TEE framework.
```
I find the "one TEE framework kind of iExec TEE tasks" confusing :thinking:

**Review by mcornaton — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -2,7 +2,12 @@
 
 ## Overview
 
-The iExec Secret Management Service (SMS) stores user secrets and provisions them to authorized Trusted Execution Environment (TEE) applications running on the iExec network.
+The _iExec Secret Management Service_ (SMS) stores user secrets and provisions them to authorized Trusted Execution Environment (TEE) applications running on the iExec network.
+
+Two TEE frameworks for TEE tasks can be supported on the iExec platform:
```
Comment (jbern0rd) [inline README.md:7]:
```suggestion
Two TEE frameworks for TEE tasks are supported on the iExec platform:
```

File: README.md [hunk 4]
```diff
@@ -16,58 +21,72 @@ The iExec Secret Management Service (SMS) stores user secrets and provisions the
...
 * A blockchain node. iExec smart contracts must be deployed on the blockchain network.
-* A Scontain Configuration and Attestation Service (CAS) to provision secrets to remote enclaves.
-* Valid OCI images configurations for pre-compute and post-compute stages of TEE tasks executions.
-* A valid OCI image configuration of a Scontain Local Attestation Service (LAS).
+* Valid OCI images configurations for pre-compute and post-compute stages of TEE tasks executions. Exposed images depend on the type of TEE framework supported.
+* A Secret Provisioner instance, in charge of provisioning secrets to remote enclaves. Each TEE framework requires its own type of Secret Provisioner.
+    * for Scone TEE tasks:
+        * a Scontain _Configuration and Attestation Service_ (CAS).
+        * a valid OCI image configuration of a Scontain Local Attestation Service (LAS).
   This service will be deployed by an iExec Worker to compute TEE tasks.
```
Comment (jbern0rd) [inline README.md:36]: Align this with LAS bullet point ?

File: README.md [hunk 5]
```diff
@@ -16,58 +21,72 @@ The iExec Secret Management Service (SMS) stores user secrets and provisions the
...
+To run properly, the _iExec Secret Management Service_ requires:
 * A blockchain node. iExec smart contracts must be deployed on the blockchain network.
-* A Scontain Configuration and Attestation Service (CAS) to provision secrets to remote enclaves.
-* Valid OCI images configurations for pre-compute and post-compute stages of TEE tasks executions.
-* A valid OCI image configuration of a Scontain Local Attestation Service (LAS).
+* Valid OCI images configurations for pre-compute and post-compute stages of TEE tasks executions. Exposed images depend on the type of TEE framework supported.
+* A Secret Provisioner instance, in charge of provisioning secrets to remote enclaves. Each TEE framework requires its own type of Secret Provisioner.
+    * for Scone TEE tasks:
+        * a Scontain _Configuration and Attestation Service_ (CAS).
+        * a valid OCI image configuration of a Scontain Local Attestation Service (LAS).
```
Comment (jbern0rd) [inline README.md:35]:
```suggestion
        * a valid OCI image configuration of a Scontain _Local Attestation Service_ (LAS).
```

File: README.md (hunk 4 above)
Comment (jeremyjams) [reply README.md]: Thank you https://github.com/iExecBlockchainComputing/iexec-sms/pull/155/commits/5695d3a7f666769e16ad41be7c9e109e333f6d47 :pray:

**Review by mcornaton — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -16,58 +21,71 @@ The iExec Secret Management Service (SMS) stores user secrets and provisions the
...
-* Valid OCI images configurations for pre-compute and post-compute stages of TEE tasks executions.
-* A valid OCI image configuration of a Scontain Local Attestation Service (LAS).
-  This service will be deployed by an iExec Worker to compute TEE tasks.
+* Valid OCI images configurations for pre-compute and post-compute stages of TEE tasks executions. Exposed images depend on the type of TEE framework supported.
+* A Secret Provisioner instance, in charge of provisioning secrets to remote enclaves. Each TEE framework requires its own type of Secret Provisioner.
+    * for Scone TEE tasks:
+        * a Scontain _Configuration and Attestation Service_ (CAS).
+        * a valid OCI image configuration of a Scontain _Local Attestation Service_ (LAS). This service will be deployed by an iExec Worker to compute TEE tasks.
+    * for Gramine TEE tasks:
+        * an _iExec SPS_ instance.
```
Comment (jbern0rd) [inline README.md:37]: For compliance with CAS, LAS, SMS, do we have the words behind the `SPS` acronym ?

**Review by jbern0rd — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (jeremyjams) [reply README.md]: Better in full yes, here it is https://github.com/iExecBlockchainComputing/iexec-sms/pull/155/commits/31db9d00a74c9bd5eb1a82d8bf4a19b54aceac11 :pray:

**Pull request merged**